redfish-core/lib/pcie.hpp

```diff
@@ -39,49 +40,93 @@ static constexpr char const* pcieDeviceInterface =
 static constexpr char const* pcieSlotInterface =
     "xyz.openbmc_project.Inventory.Item.PCIeSlot";
 
-static inline void
-    getPCIeDeviceList(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
-                      const std::string& name)
+using PCIeDevice = std::string;
```

Comment:
Why so many changes here ?

Reply:
In 1030,  PRs (e.g.  #372, #389. #391) rearranged  the codes to the most current code in this area.
One of which is to introduce - getPcieDevices(), or retrievePCIeDeviceList() etc.

Looking at them, the main restructuring can be done here via this PR so that the consecutive changes will be mostly related to the functionalities. 

For example, #389 (Link PCIeDevice to PCIeSlot) had the mix of the code restructure and functional change.
In 1050, PR #592 (i.e. #389) can be become short and simple.

Another issue which also currently exists in 1030 is to return the empty output for the first PCIe related to command after bmcweb startup.
It is due to the logic of building 'mapOfPcieDevicePaths' is done via  async_method_call as a normal function in sequence as a part of requestRoutesSystemPCIeDevice().
As the result, the very first invocation of PCIe query will only construct  mapOfPcieDevicePaths, and the the other consecutive calls are utilizing it.

This PR fixes it so that map[] will be built as a part of request...() [if map is not built] and then continue on the request function.
The other ones are capturing the code restructuring logics of 1030 PRs.